homeassistant/components/sia/alarm_control_panel.py

```diff
@@ -0,0 +1,288 @@
+
+        self._should_poll = False
+        self._is_available = True
+        self._remove_unavailability_tracker = None
+        self._state = None
+        self._old_state = None
+        self._attr = {
+            CONF_ACCOUNT: self._account,
+            CONF_PING_INTERVAL: self.ping_interval,
+            CONF_ZONE: self._zone,
+            ATTR_LAST_MESSAGE: None,
+            ATTR_LAST_CODE: None,
+            ATTR_LAST_TIMESTAMP: None,
+        }
+
+    async def async_added_to_hass(self):
+        """Once the panel is added, see if it was there before and pull in that state."""
+        await super().async_added_to_hass()
+        state = await self.async_get_last_state()
+        _LOGGER.debug(
+            "Loading last state: %s",
+            state.state if state is not None and state.state is not None else "None",
+        )
+        if (
+            state is not None
+            and state.state is not None
+            and state.state
+            in [
+                STATE_ALARM_ARMED_AWAY,
+                STATE_ALARM_ARMED_CUSTOM_BYPASS,
+                STATE_ALARM_ARMED_NIGHT,
+                STATE_ALARM_DISARMED,
+                STATE_ALARM_TRIGGERED,
+                STATE_UNKNOWN,
+            ]
+        ):
+            self.state = state.state
+        else:
+            self.state = None
+        await self._async_track_unavailable()
+        async_dispatcher_connect(
+            self.hass, DATA_UPDATED, self._schedule_immediate_update
+        )
+        self._unsub = self.hass.bus.async_listen(
+            self._event_listener_str, self.async_handle_event
+        )
+        self.async_on_remove(self._sia_on_remove)
+
+    @callback
+    def _sia_on_remove(self):
+        """Remove the unavailability and event listener."""
+        if self._unsub:
+            self._unsub()
+        if self._remove_unavailability_tracker:
+            self._remove_unavailability_tracker()
+
+    async def async_handle_event(self, event: Event):
+        """Listen to events for this port and account and update states.
+
+        If the port and account combo receives any message it means it is online and can therefore be set to available.
+        """
+        await self.assume_available()
+        if int(event.data[EVENT_ZONE]) == self._zone:
+            new_state = CODE_CONSEQUENCES.get(event.data[EVENT_CODE])
+            if new_state:
+                self._attr.update(
+                    {
+                        ATTR_LAST_MESSAGE: event.data[EVENT_MESSAGE],
+                        ATTR_LAST_CODE: event.data[EVENT_CODE],
+                        ATTR_LAST_TIMESTAMP: event.data[EVENT_TIMESTAMP],
+                    }
+                )
+                self.state = new_state
+                if not self.registry_entry.disabled:
+                    self.async_schedule_update_ha_state()
+
+    @callback
+    def _schedule_immediate_update(self):
+        self.async_schedule_update_ha_state(True)
```

Comment:
Since `async_update` isn't implemented, I don't think you need this and can simply call `self. async_write_ha_state()`